app/connectors/OttConnector.scala

```diff
@@ -18,42 +18,55 @@ package connectors
 
 import config.Service
 import models.Commodity
+import models.ott.response.OttResponse
 import play.api.Configuration
 import play.api.http.Status.{NOT_FOUND, OK}
-import play.api.libs.json.JsResult
+import play.api.libs.json.{JsResult, Reads}
 import uk.gov.hmrc.http.client.HttpClientV2
 import uk.gov.hmrc.http.{Authorization, HeaderCarrier, HttpResponse, NotFoundException, StringContextOps, UpstreamErrorResponse}
 
+import java.net.URL
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class OttConnector @Inject() (config: Configuration, httpClient: HttpClientV2)(implicit ec: ExecutionContext) {
 
-  private val baseUrl: Service              = config.get[Service]("microservice.services.online-trade-tariff-api")
-  private def ottUrl(commodityCode: String) =
+  private val baseUrl: Service                         = config.get[Service]("microservice.services.online-trade-tariff-api")
+  private def ottCommoditiesUrl(commodityCode: String) =
     url"$baseUrl/ott/commodities/$commodityCode"
 
-  def getCommodityCode(commodityCode: String)(implicit hc: HeaderCarrier): Future[Commodity] = {
+  private def ottGreenLanesUrl(commodityCode: String) =
+    url"$baseUrl/ott/goods-nomenclatures/$commodityCode"
 
-    val newHeaderCarrier = hc.copy(authorization = Some(Authorization("bearerToken")))
+  private def getFromOtt[T](commodityCode: String, urlFunc: String => URL, authToken: String)(implicit
+    hc: HeaderCarrier,
+    reads: Reads[T]
+  ): Future[T] = {
+    val newHeaderCarrier = hc.copy(authorization = Some(Authorization(authToken)))
 
     httpClient
-      .get(ottUrl(commodityCode))(newHeaderCarrier)
+      .get(urlFunc(commodityCode))(newHeaderCarrier)
       .execute[HttpResponse]
       .flatMap { response =>
         response.status match {
           case OK =>
             response.json
-              .validate[Commodity]
+              .validate[T]
               .map(result => Future.successful(result))
               .recoverTotal(error => Future.failed(JsResult.Exception(error)))
```

Comment:
can we write a test for this

Reply:
That method is private but the public usages of this higher order function are tested: getCommodityCode, and getCategorisationInfo

Reply:
If you're concerned about the upstream error, recall that there is default error handling and we shouldn't have to explicitly handle it (unless I'm mistaken)

Reply:
i suppose you could argue if we don't have a test to hit it why is it there?

Reply:
Good point, I'll have a think

Reply:
Amended it, it was working wrong anyway. Added in a test, coverage should be fine now.